src/ipc/StoreMap.h

```diff
@@ -262,6 +262,9 @@ class StoreMap
     /// only works on locked entries; returns nil unless the slice is readable
     const Anchor *peekAtReader(const sfileno fileno) const;
 
+    /// only works on locked entries; returns nil unless the slice is writable
```

Comment:
This is incomplete. The comment inside the definition indicates that this may *only* be called by the lock holder itself. Other objects/code/threads/workers may not call this method regardless of its lock status.

Reply:
There may be several lock holders: a writer and some readers. The description explains that this method may be called only _after_ the entry was opened for writing or reading, i.e., acquired such a lock.

Reply:
> Amos: Other objects/code/threads/workers may not call this method regardless of its lock status.

What you are saying is essentiallyIn SMP world, one cannot tell whether an entry is locked without locking it... implied by the existing description: In SMP world, one cannot tell whether an entry is locked without locking it. Thus, "other objects/code/threads/workers" (i.e. those without a lock) must not call this method because it only works on locked entries, and those "others" _cannot tell_ whether the entry is locked. Those others can only check whether the entry was locked at some point in the past, and that is not enough to satisfy this method precondition.

If you prefer, we can replace "only works on locked entries" with "the caller must hold a lock on the entry" or something like that. This replacement should be done for both peekAt*() methods. Due to its length, it should probably be moved to a second description line. @yadij, would this replacement address your concern?

Reply:
> Those others can only check whether the entry was locked at some point in the past, and that is not enough to satisfy this method precondition.

That is the answer to my question then. Only the lock holder may call this.

> 
```suggestion
    /// the caller must hold a lock on the entry
    /// \returns nullptr unless the slice is writable
```

Reply:
Done.